app/assets/javascripts/components/activity/activity_table.cjsx

```diff
@@ -39,9 +43,9 @@ ActivityTable = React.createClass(
 
     drawers = @state.activity.map (revision) ->
       courses = revision.courses.map (course) ->
-        <li><a href="/courses/#{course.slug}">{course.title}</a></li>
+        <li key={Math.random() * course.id}><a href="/courses/#{course.slug}">{course.title}</a></li>
 
-      <tr className='activity-table-drawer'>
+      <tr key={Math.random() * revision.key} className='activity-table-drawer'>
```

Comment:
Keyed elements in React can be reused when the page rerenders; if the keys change every time the page rerenders, then it will keep creating new elements instead of reusing existing ones.

It's probably not really a performance issue, but if any of the components with randomized keys have their own state (for example, an open/closed state, or some other state related to behavior we haven't built yet) then that state would be lost on every render, because the state is tied to the key.

https://coderwall.com/p/jdybeq/the-importance-of-component-keys-in-react-js